components/support/guid/src/lib.rs

```diff
@@ -442,9 +442,12 @@ mod test {
         );
 
         // order by data instead of length
-        assert!(Guid::from("zzz") > Guid::from("aaaaaa"));
-        assert!(Guid::from("ThisIsASolowGuid") < Guid::from("zzz"));
-        assert!(Guid::from("ThisIsASolowGuid") > Guid::from("AnotherSlowGuid"));
+        // hrmph - clippy in 1.54-nightly wanted the original, below:
+        // > assert!(Guid::from("zzz") > Guid::from("aaaaaa"));
+        // changed to below: what magic is this? :/
+        assert!(*"zzz" > *"aaaaaa");
```

Comment:
this freaks me out though!

Reply:
This PR looks like a really nice collection of cleanups and simplifications...and then WHAT THE WHAT SORCERY IS THIS I DONT EVEN UNDERSTAND WHAT THIS IS SUPPOSED TO BE DOING. This can't possibly be right can it?

Reply:
I think clippy has mis-fired here. If I change the `impl Ord for Guid` to switch the order of comparison, then the old-style tests correctly started failing, while the new-style tests continue to pass. We should revert this IMHO.

Reply:
yeah, agreed. Even if it was correct I was going to suggest we just allow that lint

Reply:
The alleged failing lint here is [`cmp_owned`](https://rust-lang.github.io/rust-clippy/master/#cmp_owned) which seems to have [a little bit of a history of false positives](https://github.com/rust-lang/rust-clippy/search?q=cmp_owned&type=issues).